pyro/nn/auto_reg_nn.py

```diff
@@ -96,22 +97,25 @@ def forward(self, _input):
         return F.linear(_input, masked_weight, self.bias)
 
 
-class AutoRegressiveNN(nn.Module):
+class ConditionalAutoRegressiveNN(nn.Module):
     """
-    An implementation of a MADE-like auto-regressive neural network.
+    An implementation of a MADE-like auto-regressive neural network that can condition on an additional input.
 
     Example usage:
 
     >>> x = torch.randn(100, 10)
-    >>> arn = AutoRegressiveNN(10, [50], param_dims=[1])
-    >>> p = arn(x)  # 1 parameters of size (100, 10)
-    >>> arn = AutoRegressiveNN(10, [50], param_dims=[1, 1])
-    >>> m, s = arn(x) # 2 parameters of size (100, 10)
-    >>> arn = AutoRegressiveNN(10, [50], param_dims=[1, 5, 3])
-    >>> a, b, c = arn(x) # 3 parameters of sizes, (100, 1, 10), (100, 5, 10), (100, 3, 10)
+    >>> y = torch.randn(100, 5)
+    >>> arn = ConditionalAutoRegressiveNN(10, 5, [50], param_dims=[1])
+    >>> p = arn(x, context=y)  # 1 parameters of size (100, 10)
+    >>> arn = ConditionalAutoRegressiveNN(10, 5, [50], param_dims=[1, 1])
+    >>> m, s = arn(x, context=y) # 2 parameters of size (100, 10)
+    >>> arn = ConditionalAutoRegressiveNN(10, 5, [50], param_dims=[1, 5, 3])
+    >>> a, b, c = arn(x, context=y) # 3 parameters of sizes, (100, 1, 10), (100, 5, 10), (100, 3, 10)
 
-    :param input_dim: the dimensionality of the input
+    :param input_dim: the dimensionality of the input variable
     :type input_dim: int
+    :param context_dim: the dimensionality of the context variable
+    :type context_dim: int
```

Comment:
nit: it would be clearer if you add a reference to https://arxiv.org/abs/1602.06701.